src/proxy/main.c

```diff
@@ -0,0 +1,118 @@
+}
+
+int parse_node_unit_opt(const char *opt_node_unit, UNUSED char **ret_node_name, UNUSED char **ret_unit_name) {
+        if (opt_node_unit == NULL || ret_node_name == NULL || ret_unit_name == NULL) {
+                return -EINVAL;
+        }
+
+        // copy the option string as strtok_r manipulates it
+        char *rest = strdup(opt_node_unit);
+        if (rest == NULL) {
+                return -ENOMEM;
+        }
+
+        _cleanup_free_ char *token = NULL;
+        token = strtok_r(rest, "__", &rest);
+        *ret_node_name = strdup(token);
+        *ret_unit_name = strdup(strtok_r(rest, "__", &rest));
+
+        return 0;
+}
+
+static int get_opts(int argc, char *argv[]) {
+        int opt = 0;
+        while ((opt = getopt_long(argc, argv, OPTIONS_STR, options, NULL)) != -1) {
+                if (opt == ARG_HELP_SHORT) {
+                        usage(argv);
+                        return 0;
+                }
+
+                fprintf(stderr, "Unsupported option %c\n", opt);
+                usage(argv);
+                return -EINVAL;
+        }
+
+        if (argc < 2) {
+                return -EINVAL;
+        }
+        opt_node_unit = argv[1];
+
+        return 0;
+}
+
+int main(int argc, char *argv[]) {
+        hirte_log_init();
+        hirte_log_set_log_fn(hirte_log_to_stderr_with_location);
+
+        int r = get_opts(argc, argv);
+        if (r < 0) {
+                return EXIT_FAILURE;
+        }
+        if (opt_node_unit == NULL) {
+                return EXIT_SUCCESS;
+        }
+
+        _cleanup_free_ char *node_name = NULL;
+        _cleanup_free_ char *unit_name = NULL;
+        parse_node_unit_opt(opt_node_unit, &node_name, &unit_name);
+
+        _cleanup_sd_event_ sd_event *event = NULL;
+        r = sd_event_default(&event);
+        if (r < 0) {
+                hirte_log_errorf("Failed to create event loop: %s", strerror(-r));
+                return EXIT_FAILURE;
+        }
+
+        _cleanup_sd_bus_ sd_bus *api_bus = NULL;
+#ifdef USE_USER_API_BUS
+        api_bus = user_bus_open(event);
+#else
+        api_bus = system_bus_open(event);
+#endif
+        if (api_bus == NULL) {
+                hirte_log_error("Failed to create api bus");
+                return EXIT_FAILURE;
+        }
+
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *message = NULL;
+        r = sd_bus_call_method(
+                        api_bus,
+                        HIRTE_AGENT_DBUS_NAME,
+                        HIRTE_AGENT_OBJECT_PATH,
+                        AGENT_INTERFACE,
+                        "CreateProxy",
+                        &error,
+                        &message,
+                        "sss",
+                        node_name,
+                        unit_name,
+                        "replace");
+        if (r < 0) {
+                hirte_log_errorf("Failed to create proxy: %s", strerror(-r));
```

Comment:
I don't think it is the right thing to log an error here. This will return an error if the remote service failed to start for whatever reason. I mean, this is an error in some sense, but it is still expected and natural for the proxy service that this happens.

Maybe we can report a specific error code for that case and just log an error for other error codes.